pkg/secrets/pushtofile/retrieve_secrets_test.go

```diff
@@ -0,0 +1,122 @@
+package pushtofile
+
+import (
+	"reflect"
+	"testing"
+
+	conjurMocks "github.com/cyberark/secrets-provider-for-k8s/pkg/secrets/clients/conjur/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+type retrieveSecretsTestCase struct {
+	description string
+	secretSpecs map[string][]SecretSpec
+	assert      func(t *testing.T, result SecretGroups, err error)
+}
+
+func createSecretGroups(groupSpecs map[string][]SecretSpec) SecretGroups {
+	secretGroups := SecretGroups{}
+	for label, secretSpecs := range groupSpecs {
+		secretGroup := SecretGroup{
+			Label:         label,
+			SecretSpecs:   secretSpecs,
+			SecretsValues: map[string]string{},
+		}
+		secretGroups = append(secretGroups, secretGroup)
+	}
+	return secretGroups
+}
+
+func findGroupValues(secretGroups SecretGroups, label string) *map[string]string {
+	for _, secretGroup := range secretGroups {
+		if secretGroup.Label == label {
+			return &secretGroup.SecretsValues
+		}
+	}
+	return nil
+}
+
+func valuesAreEqual(values1, values2 map[string]string) bool {
+	return reflect.DeepEqual(values1, values2)
+}
+
+func assertGoodResults(expectedGroupValues map[string]map[string]string) func(*testing.T, SecretGroups, error) {
+	return func(t *testing.T, result SecretGroups, err error) {
+
+		if !assert.NoError(t, err) {
+			return
+		}
+		for groupLabel, expValues := range expectedGroupValues {
+			actualValues := findGroupValues(result, groupLabel)
+			assert.NotNil(t, actualValues)
+			assert.True(t, valuesAreEqual(*actualValues, expValues))
+		}
+	}
+}
+
+var retrieveSecretsTestCases = []retrieveSecretsTestCase{
+	{
+		description: "Happy Case",
+		secretSpecs: map[string][]SecretSpec{
+			"cache": []SecretSpec{
+				{Alias: "api-url", Id: "dev/openshift/api-url"},
+				{Alias: "username", Id: "dev/openshift/username"},
+				{Alias: "password", Id: "dev/openshift/password"},
+			},
+			"db": []SecretSpec{
+				{Alias: "api-url", Id: "ci/openshift/api-url"},
+				{Alias: "username", Id: "ci/openshift/username"},
+				{Alias: "password", Id: "ci/openshift/password"},
+			},
+		},
+		assert: assertGoodResults(map[string]map[string]string {
```

Comment:
Nice! I like the format of the test cases.  Assertions on input-output pairs for the happy path, followed by deviation from the happy path.